api/apps/api/src/modules/projects/job-status/job-status.view.api.entity.ts

```diff
@@ -0,0 +1,76 @@
+import { ViewColumn, ViewEntity } from 'typeorm';
+import { JobType } from '@marxan-api/modules/projects/job-status/jobs.enum';
+import { JobStatus } from '@marxan-api/modules/scenarios/scenario.api.entity';
+import { API_EVENT_KINDS } from '@marxan/api-events';
+
+@ViewEntity({
+  expression: `
+    SELECT
+      DISTINCT ON ("jobType", topic) "jobType",
+      api_events.topic AS "scenarioId",
+      projects.id AS "projectId",
+      api_events.kind
+    FROM
+      api_events
+      INNER JOIN scenarios ON api_events.topic = scenarios.id
+      INNER JOIN projects ON projects.id = scenarios.project_id
+      CROSS JOIN LATERAL SUBSTRING(
+        api_events.kind
+        FROM
+          'scenario.#"[^.]*#"%' FOR '#'
```

Comment:
That looks scary, what it exactly matches?

Reply:
it's just a regular expression
https://www.postgresql.org/docs/9.3/functions-matching.html#FUNCTIONS-SIMILARTO-REGEXP
matches everything what is not a dot after `scenario.` and substrings it